File: nxflix-agents-python/src/nxflix_agents/models/study.py
"""Study recommendation and session request models."""

from pydantic import BaseModel, Field


class StudyRecommendation(BaseModel):
    """A personalized study recommendation."""

    grammar_ids: list[str]
    reason: str
    priority: int = Field(default=1, ge=1, le=5)
    estimated_time_minutes: int = 15
    focus_areas: list[str] = []
    suggested_question_types: list[str] = []


class RecommendationRequest(BaseModel):
    """Request for study recommendations."""

    user_id: str
    available_grammar_ids: list[str] = []
    max_items: int = Field(default=5, ge=1, le=20)
    focus_weak_areas: bool = True
    include_new: bool = True
    time_available_minutes: int | None = None


class SessionRequest(BaseModel):
    """Request to start a study session."""

    user_id: str
    grammar_ids: list[str]
    question_count: int = Field(default=10, ge=1, le=50)
    question_types: list[str] = []
    difficulty: int | None = Field(default=None, ge=1, le=5)


class SessionCompleteRequest(BaseModel):
    """Request to complete a study session."""

    session_id: str
    results: list[dict]  # List of SessionResult-like dicts

File: nxflix-agents-ts/src/models/study.ts
import { z } from 'zod';

export const StudyRecommendation = z.object({
  grammarIds: z.array(z.string()),
  reason: z.string(),
  priority: z.number().int().min(1).max(5).default(1),
  estimatedTimeMinutes: z.number().int().default(15),
  focusAreas: z.array(z.string()).default([]),
  suggestedQuestionTypes: z.array(z.string()).default([]),
});
export type StudyRecommendation = z.infer<typeof StudyRecommendation>;

export const RecommendationRequest = z.object({
  userId: z.string(),
  availableGrammarIds: z.array(z.string()).default([]),
  maxItems: z.number().int().min(1).max(20).default(5),
  focusWeakAreas: z.boolean().default(true),
  includeNew: z.boolean().default(true),
  timeAvailableMinutes: z.number().int().nullish(),
});
export type RecommendationRequest = z.infer<typeof RecommendationRequest>;

export const SessionRequest = z.object({
  userId: z.string(),
  grammarIds: z.array(z.string()),
  questionCount: z.number().int().min(1).max(50).default(10),
  questionTypes: z.array(z.string()).default([]),
  difficulty: z.number().int().min(1).max(5).nullish(),
});
export type SessionRequest = z.infer<typeof SessionRequest>;

export const SessionCompleteRequest = z.object({
  sessionId: z.string(),
  results: z.array(z.record(z.unknown())),
});
export type SessionCompleteRequest = z.infer<typeof SessionCompleteRequest>;
